spec/std/crystal/compiler_rt/muloti4_spec.cr

```diff
@@ -0,0 +1,95 @@
+  test__muloti4(0_i128, 0_i128, 0_i128, 0)
+  test__muloti4(0_i128, 1_i128, 0_i128, 0)
+  test__muloti4(1_i128, 0_i128, 0_i128, 0)
+  test__muloti4(0_i128, 10_i128, 0_i128, 0)
+  test__muloti4(10_i128, 0_i128, 0_i128, 0)
+  test__muloti4(0_i128, 81985529216486895_i128, 0_i128, 0)
+  test__muloti4(81985529216486895_i128, 0, 0, 0)
+
+  test__muloti4(0_i128, Int128RT[1_i128].negate!, 0_i128, 0)
+  test__muloti4(Int128RT[1_i128].negate!, 0_i128, 0_i128, 0)
+  test__muloti4(0_i128, Int128RT[10_i128].negate!, 0_i128, 0)
+  test__muloti4(Int128RT[10_i128].negate!, 0_i128, 0_i128, 0)
+  test__muloti4(0_i128, Int128RT[81985529216486895_i128].negate!, 0_i128, 0)
+  test__muloti4(Int128RT[81985529216486895_i128].negate!, 0_i128, 0_i128, 0)
+
+  test__muloti4(1_i128, 1_i128, 1_i128, 0)
+  test__muloti4(1_i128, 10_i128, 10_i128, 0)
+  test__muloti4(10_i128, 1_i128, 10_i128, 0)
+  test__muloti4(1_i128, 81985529216486895_i128, 81985529216486895_i128, 0)
+  test__muloti4(81985529216486895_i128, 1_i128, 81985529216486895_i128, 0)
+
+  test__muloti4(1_i128, Int128RT[1_i128].negate!, Int128RT[1_i128].negate!, 0)
+  test__muloti4(1_i128, Int128RT[10_i128].negate!, Int128RT[10_i128].negate!, 0)
+  test__muloti4(Int128RT[10_i128].negate!, 1_i128, Int128RT[10_i128].negate!, 0)
+  test__muloti4(1_i128, Int128RT[81985529216486895_i128].negate!, Int128RT[81985529216486895_i128].negate!, 0)
+  test__muloti4(Int128RT[81985529216486895_i128].negate!, 1_i128, Int128RT[81985529216486895_i128].negate!, 0)
+
+  test__muloti4(3037000499_i128, 3037000499_i128, 9223372030926249001_i128, 0)
+  test__muloti4(Int128RT[3037000499_i128].negate!, 3037000499_i128, Int128RT[9223372030926249001_i128].negate!, 0)
+  test__muloti4(3037000499_i128, Int128RT[3037000499_i128].negate!, Int128RT[9223372030926249001_i128].negate!, 0)
+  test__muloti4(Int128RT[3037000499_i128].negate!, Int128RT[3037000499_i128].negate!, 9223372030926249001_i128, 0)
+
+  test__muloti4(4398046511103_i128, 2097152_i128, 9223372036852678656_i128, 0)
+  test__muloti4(Int128RT[4398046511103_i128].negate!, 2097152_i128, Int128RT[9223372036852678656_i128].negate!, 0)
+  test__muloti4(4398046511103_i128, Int128RT[2097152_i128].negate!, Int128RT[9223372036852678656_i128].negate!, 0)
+  test__muloti4(Int128RT[4398046511103_i128].negate!, Int128RT[2097152_i128].negate!, 9223372036852678656_i128, 0)
+
+  test__muloti4(2097152_i128, 4398046511103_i128, 9223372036852678656_i128, 0)
+  test__muloti4(Int128RT[2097152_i128].negate!, 4398046511103_i128, Int128RT[9223372036852678656_i128].negate!, 0)
+  test__muloti4(2097152_i128, Int128RT[4398046511103_i128].negate!, Int128RT[9223372036852678656_i128].negate!, 0)
+  test__muloti4(Int128RT[2097152_i128].negate!, Int128RT[4398046511103_i128].negate!, 9223372036852678656_i128, 0)
+
+  # test__muloti4(Int128RT[0x00000000000000B5_i64, 0x04F333F9DE5BE000_u64], Int128RT[0x0000000000000000_i64, 0x00B504F333F9DE5B_u64], Int128RT[0x7FFFFFFFFFFFF328_i64, 0xDF915DA296E8A000_u64], 0)
```

Comment:
Why are there so many commented out tests?

Reply:
they are broken.  The main problem is `__udivmodti4` but it is hard to debug and I work on one failure at a time.  I have gotten enough help from these PRs that I can close them and reopen them when I am closer if that helps.

Reply:
If this is WIP, then can you indicate that in the title?